_rules/active-focus-shortcut-1e9941.md

```diff
@@ -0,0 +1,212 @@
+---
+id: 1e9941
+name: Keyboard shortcut is only active when component has focus
+rule_type: atomic
+description: |
+  This rule checks that if keyboard shortcuts are implemented using only printable characters for a user interface component, then they are only available when that component has focus.
+accessibility_requirements: # Remove whatever is not applicable
+  wcag21:2.1.4: # Character Key Shortcuts (A)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgements:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [HTML document][] with [keyboard shortcuts][keyboard shortcuts].
+
+## Expectation
+
+For each [user interface component][] that is a [descendent][] of the root node of the test target, if [keyboard shortcuts][] are implemented using only [printable characters][], then they are only available when that component has focus.
+
+## Assumptions
+
+This rule assumes as applicable [keyboard shortcuts][keyboard shortcuts] those implemented by the test target [content][]. Any other means (e.g. browser extensions, browser settings, user agents, external browser applications) are not considered.
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 2.1.4: Character Key Shortcuts](https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html)
+- [G127 Providing a mechanism to allow users to remap or turn off character key shortcuts](https://www.w3.org/WAI/WCAG21/Techniques/general/G217)
+- [F99 Failure of Success Criterion 2.1.4 due to implementing character key shortcuts that cannot be turned off or remapped](https://www.w3.org/WAI/WCAG21/Techniques/failures/F99)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1

```

Comment:
I'm still wondering whether Passed Examples 1 and 2 could use the same "add to list" code to avoid having to understand too many things. `ctrl+'+'` for adding to list, or a button with acces key and `+` should both be perfectly fine examples.

Reply:
Not sure I understand what you mean here

Reply:
I'm thinking that all examples could use the "add to list" example (with a `ctrl+'+'` shortcut). What is added to the list is not really important, so even if it's not an input field it's OK (it could be a random number, for example).

That would make a single example (and code) to understand. That would also allow factoring the code out in a test assets.

Reply:
OK. Now the refactoring makes more sense (because refactoring just the list example, while the other examples remained the same would results in increased effort for the reader, I believe). I'll try to find some time to work on this 👍